pkg/v1/tkg/tkgctl/create_cluster_test.go

```diff
@@ -327,7 +420,205 @@ var _ = Describe("Unit tests for - ccluster.yaml as input file for 'tanzu cluste
+			Expect("3").To(Equal(fmt.Sprintf("%v", mappedVal)))
+			// "spec.topology.variables.controlPlane.machine.numCPUs : ConfigVariableVsphereCPNumCpus
+			mappedVal, _ = ctl.TKGConfigReaderWriter().Get(constants.ConfigVariableVsphereCPNumCpus)
+			Expect("2").To(Equal(fmt.Sprintf("%v", mappedVal)))
+
+			// "spec.topology.variables.node.machine.memoryMiB": ConfigVariableVsphereWorkerMemMib,
+			mappedVal, _ = ctl.TKGConfigReaderWriter().Get(constants.ConfigVariableVsphereWorkerMemMib)
+			Expect("16384").To(Equal(fmt.Sprintf("%v", mappedVal)))
+
+			// "spec.topology.workers.machineDeployments.0.replicas": ConfigVariableWorkerMachineCount
+			mappedVal, _ = ctl.TKGConfigReaderWriter().Get(constants.ConfigVariableWorkerMachineCount)
+			Expect("4").To(Equal(fmt.Sprintf("%v", mappedVal)))
+
+			// TopologyWorkersMachineDeploymentsFailureDomain2: ConfigVariableVsphereAz2
+			mappedVal, _ = ctl.TKGConfigReaderWriter().Get(constants.ConfigVariableVsphereAz2)
+			Expect("3").To(Equal(fmt.Sprintf("%v", mappedVal)))
+
+		})
+	})
+})
+
+var _ = Describe("Unit tests for - (Azure) - cluster_azure.yaml as input file for 'tanzu cluster create -f cluster_azure.yaml' use case", func() {
+	var (
+		ctl       tkgctl
+		tkgClient = &fakes.Client{}
+		bomClient = &fakes.TKGConfigBomClient{}
+		options   CreateClusterOptions
+	)
+	JustBeforeEach(func() {
+		tkgConfigReaderWriter, _ := tkgconfigreaderwriter.NewReaderWriterFromConfigFile(configFilePath, configFilePath)
+		ctl = tkgctl{
+			configDir:              testingDir,
+			tkgClient:              tkgClient,
+			tkgConfigReaderWriter:  tkgConfigReaderWriter,
+			tkgConfigUpdaterClient: tkgconfigupdater.New(testingDir, nil, tkgConfigReaderWriter),
+			tkgBomClient:           bomClient,
+		}
+		tkgClient.IsFeatureActivatedReturns(true)
+	})
+	Context("'tanzu cluster create -f cluster_azure.yaml' use case - vsphere provider - devcc plan:", func() {
+		BeforeEach(func() {
+			options = CreateClusterOptions{
+				ClusterName:            "test-cluster",
+				Plan:                   "dev",
+				InfrastructureProvider: "",
+				Namespace:              "",
+				GenerateOnly:           false,
+				TkrVersion:             fakeTKRVersion,
+				SkipPrompt:             true,
+				Edition:                "tkg",
+				ClusterConfigFile:      cclassInputFileAzure,
+			}
+		})
+		It("'tanzu cluster create -f cluster_azure.yaml' use case - vsphere provider - devcc plan:, make sure legacy configurations are updated..", func() {
+			/*
+			 Process input ccluster.yaml file, this should process input ccluster.yaml file
+			 and update the environment with legacy name and values
+			 most of ccluster.yaml attributes are mapped to legacy variable for more look this - constants.CclusterToLegacyVariablesMapAzure
+			*/
+			IsInputFileClusterClassBased, err := ctl.processWorkloadClusterInputFile(&options)
+			Expect(IsInputFileClusterClassBased).Should(BeTrue())
+			Expect(err).To(BeNil())
+
+			/*
```

Comment:
nit:
probably 
```
// xx
// xx 
```
style works just as well and more aligned with other comments in the file. Also the indenting is inconsistent across different blocks

Reply:
I have updated all comments with uniform indentation. More than two lines of comments, I have updated as multiline comments otherwise with single-line comments.